### PR TITLE
Change Google Analytics to Apache Matomo instance

### PR DESCRIPTION
**Pull request by tuhaihe**
closes: #202 

---

## Change logs

As the ASF website checks and the privacy policy, we're not encouraged to use Google Analytics for our website. So change it to Matomo.

Also, delete the unused Google Anlytics-related modules from the pacakge list.

## Contributor's checklist

Here are some reminders before you submit your pull request:

* Make sure that your Pull Request has a clear title and commit message. You can take the [Git commit template](https://github.com/apache/cloudberry/blob/main/.gitmessage) as a reference.
* Learn the [code contribution](https://cloudberry.apache.org/contribute/code) and [doc contribution](https://cloudberry.apache.org/contribute/doc) guides for better collaboration.
* Make sure that your changes deployment preview is successful.
* List your communications in the [GitHub Issues](https://github.com/apache/cloudberry-site/issues) or [Discussions](https://github.com/apache/cloudberry/discussions) (if has or needed).
* Feel free to ask for the [cloudberry committers](https://github.com/orgs/apache/teams/cloudberry-committers) or other people to help review and approve.

### CODE REVIEW

File: docusaurus.config.ts [hunk 1]
```diff
@@ -277,6 +274,13 @@ const config: Config = {
       theme: prismThemes.github,
       darkTheme: prismThemes.dracula,
     },
+    // Change website analytics from Google Analytics to Matomo
+    matomo: {
+      matomoUrl: 'https://analytics.apache.org/',
+      siteId: 'ID',
```
Comment (tuhaihe) [inline docusaurus.config.ts:280]: Now have asked for one Site ID from the ASF Privacy team. Once got it, will update this field.

**Comment by tuhaihe** [comment]
After merging this PR, this link will provide access to the site statistics: https://analytics.apache.org/index.php?module=CoreHome&action=index&date=yesterday&period=day&idSite=66#?period=day&date=yesterday&category=Dashboard_Dashboard&subcategory=1

**Comment by edespino** [comment]
@tuhaihe Are you planning on adding a project based information (including links) for Cloudberry's Matomo analytics page somewhere? I think it should be readily accessible by the dev community. Any details on it's setup, configuration and administration (or pointers to information with this) might also be helpful.

I have been thinking of this from the developer and release perspective. With Apache MADlib we used the Apache Wiki. As a project we have the option of enabling the GitHub wiki feature for such dynamic information. I have a need to capture project release engineering based information and I would prefer an easy and collaborative space and thus I am leaning toward the GitHub wiki. What do you think?

**Comment by tuhaihe** [comment]
> @tuhaihe Are you planning on adding a project based information (including links) for Cloudberry's Matomo analytics page somewhere? I think it should be readily accessible by the dev community. Any details on it's setup, configuration and administration (or pointers to information with this) might also be helpful.
> 
> I have been thinking of this from the developer and release perspective. With Apache MADlib we used the Apache Wiki. As a project we have the option of enabling the GitHub wiki feature for such dynamic information. I have a need to capture project release engineering based information and I would prefer an easy and collaborative space and thus I am leaning toward the GitHub wiki. What do you think?

Good idea. We can also create a new page module named `Developer Guide` on the website to help community developers learn about the development aspects of the kernel, releases, website, and other areas, which will be easier to maintain and search for by developers and search engines like Google.

Does it make sense to you? I would like to create one new PR for this.

**Comment by edespino** [comment]
> > @tuhaihe Are you planning on adding a project based information (including links) for Cloudberry's Matomo analytics page somewhere? I think it should be readily accessible by the dev community. Any details on it's setup, configuration and administration (or pointers to information with this) might also be helpful.
> > I have been thinking of this from the developer and release perspective. With Apache MADlib we used the Apache Wiki. As a project we have the option of enabling the GitHub wiki feature for such dynamic information. I have a need to capture project release engineering based information and I would prefer an easy and collaborative space and thus I am leaning toward the GitHub wiki. What do you think?
> 
> Good idea. We can also create a new page module named `Developer Guide` on the website to help community developers learn about the development aspects of the kernel, releases, website, and other areas, which will be easier to maintain and search for by developers and search engines like Google.
> 
> Does it make sense to you? I would like to create one new PR for this.

Cool ... Maybe we can create a basic top-level page "Project and Engineering Guides" with sub-pages. Here is a basic break down we use to help us think through this.

Infrastructure
 - Website
 - Metrics
 - Bots
 - DockerHub
 
Social Media
- Accounts

Release Engineering
- Release numbering scheme
- Release cadence
- Release process

Development guides
- basic development
- test execution
- Deep dives
  - kernel
  - query optimizers
- components
- GitHub CI use

**Review by edespino — APPROVED** [review]
LGTM

**Pull request merged**